subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/AnnotationProcessorPathFactory.java

```diff
@@ -31,29 +31,23 @@
 import java.util.Collections;
 import java.util.LinkedHashSet;
 import java.util.List;
-import java.util.Map;
 import java.util.Set;
 
 
 public class AnnotationProcessorPathFactory {
```

Comment:
This whole class should go. The annotationProcessorPath should always be assigned, never null and directly passed on.

Reply:
There's another task to remove support for `-processorpath` (https://github.com/gradle/gradle/issues/6295). However, using just `-processor` wasn't deprecated so we cannot remove it, can we?

Reply:
Not sure what you mean -processor doesn't specify a path. It just says which classes from the path to use. We don't have to do anything special for it anymore.

Reply:
I meant the scenario of using `-processor` without using the `-processorpath` option or the `annotationProcessor` configuration. Currently, we fallback to using the `classpath` for that and we don't emit a deprecation warning. Do you think we can break that?

Reply:
Yes, we should.

Reply:
Alright, I've removed falling back to `compileClasspath` when `-processor` is set and `annotationProcessorPath` is empty in 0285f2120e9. I'll remove reading `-processorpath` in a separate PR.